src/Sulu/Bundle/ContentBundle/Collaboration/CollaborationMessageHandler.php

```diff
@@ -0,0 +1,211 @@
+ *
+ * The example calls the enter action and passes the current page and the user id 
+ * as message parameter
+ */
+class CollaborationMessageHandler implements MessageHandlerInterface
+{
+    /**
+     * @var UserRepositoryInterface
+     */
+    private $userRepository;
+
+    /**
+     * An array which contains all the collaborators for a certain identifier (representing a page, product, ...).
+     *
+     * @var Collaborator[][]
+     */
+    private $collaborators = [];
+
+    public function __construct($userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(ConnectionInterface $conn, array $message, MessageHandlerContext $context)
+    {
+        try {
+            return $this->execute($conn, $context, $message);
+        } catch (\Exception $e) {
+            throw new MessageHandlerException($e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function onClose(ConnectionInterface $conn, MessageHandlerContext $context)
+    {
+        $this->leave($conn, $context, ['command' => 'leave']);
+    }
+
+    /**
+     * Executes command.
+     *
+     * @param ConnectionInterface $conn
+     * @param MessageHandlerContext $context
+     * @param array $msg
+     *
+     * @return mixed|null
+     *
+     * @throws MissingParameterException
+     */
+    private function execute(ConnectionInterface $conn, MessageHandlerContext $context, $msg)
+    {
+        if (!array_key_exists('command', $msg)) {
+            throw new MissingParameterException('command');
+        }
+        $command = $msg['command'];
+        $result = null;
+
+        switch ($command) {
+            case 'enter':
+                $result = $this->enter($conn, $context, $msg);
+                break;
+            case 'leave':
+                $result = $this->leave($conn, $context, $msg);
+                break;
+            default:
+                throw new \InvalidArgumentException(sprintf('Command "%s" not known', $command));
+                break;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Called when the user has entered the page.
+     *
+     * @param ConnectionInterface $conn
+     * @param MessageHandlerContext $context
+     * @param array $msg
+     *
+     * @return array
+     */
+    private function enter(ConnectionInterface $conn, MessageHandlerContext $context, $msg)
+    {
+        $identifier = $this->getUniqueCollaborationIdentifier($msg);
+        $user = $this->userRepository->findUserById($msg['userId']);
+        $this->addCollaborator($conn, $identifier, $user);
+
+        $users = array_map(
+            function (Collaborator $collaborator) {
+                return [
+                    'id' => $collaborator->getUser()->getId(),
+                    'username' => $collaborator->getUser()->getUsername(),
+                    'fullName' => $collaborator->getUser()->getFullName(),
+                ];
```

Comment:
I'm not the biggest fan of the serializer, so for me this is fine. If you are going to use the serializer I would suggest `minimal` and skip the user suffix.